src/solver/optimisation/adequacy_patch_csr/set_problem_cost_function.cpp

```diff
@@ -49,7 +49,7 @@ double calculateQuadraticCost(const PROBLEME_HEBDO* problemeHebdo, int hour, int
     else if (problemeHebdo->adqPatchParams->PriceTakingOrder
              == Data::AdequacyPatch::AdqPatchPTO::isDens)
     {
-        priceTakingOrders = problemeHebdo->ResultatsHoraires[area]->ValeursHorairesDENS[hour];
+        priceTakingOrders = problemeHebdo->ResultatsHoraires[area]->ValeursHorairesDENS[hour]; // densNew value taken! OK!
```

Comment:
Careful, `ValeursHorairesDENS[hour] = 0`, see **src/solver/optimisation/adequacy_patch_local_matching/adequacy_patch_weekly_optimization.cpp**.

```suggestion
        priceTakingOrders = problemeHebdo->ResultatsHoraires[pays]->ValeursHorairesDeDefaillancePositive[hour];
```

Reply:
I'm positive we are good here. According to spec:
![image](https://user-images.githubusercontent.com/97689304/222175064-999c36a1-8e42-4e87-8824-11a09c518555.png)
s0 PTO=DENS not PTO=ENS.
if ValeursHorairesDENS[hour] = 0 function calculateQuadraticCost will just return zero. It will not try to divide with zero.
If you meant that by this point in the code the column DENS is going to be all zeros, It is not the case. Zeros are overwritten for areas inside adq-patch in 
src/solver/optimisation/post_process_commands.cpp line 231

Reply:
Ok, LGTM